webservices/legal_docs/advisory_opinions.py

```diff
@@ -14,24 +14,25 @@
 
 ALL_AOS = """
     SELECT
-        ao_id,
-        ao_no,
-        name,
-        summary,
-        req_date,
-        issue_date,
-        CASE WHEN finished IS NULL THEN TRUE ELSE FALSE END AS is_pending
-    FROM aouser.aos_with_parsed_numbers ao
-    LEFT JOIN (SELECT DISTINCT ao_id AS finished
-               FROM aouser.document
-               WHERE category IN ('Final Opinion', 'Withdrawal of Request')) AS finished
-        ON ao.ao_id = finished.finished
+        ao_parsed.ao_id,
+        ao_parsed.ao_no,
+        ao_parsed.name,
+        ao_parsed.summary,
+        ao_parsed.req_date,
+        ao_parsed.issue_date,
+        CASE ao.stage WHEN 2 THEN 'Withdrawn'
```

Comment:
@vrajmohan I removed the is_pending logic here, added the function on line 85, and added the is_pending logic on line 117.

Reply:
That's great! Sorry, I wasn't clear but ```CASE ao.stage WHEN 2 THEN 'Withdrawn'
            --Hard-code one withdrawn AO that has improperly been marked Final
            WHEN 1 THEN CASE ao.ao_no WHEN '2009-05' THEN 'Withdrawn' ELSE 'Final' END
            ELSE 'Pending' END AS stage``` would also be more clear in `python`.

Reply:
To elaborate, the function would be:
```
def ao_stage_to_status(ao_no, stage):
   AOS_WITH_INCORRECT_STAGE = ['2009-05']
   if stage == 2:
       return "Withdrawn"
   elif stage == 1:
       if ao_no in AOS_WITH_INCORRECT_STAGE:
           return "Withdrawn:
       else:
           return "Final"
  else:
      return "Pending"
```
This would make it easier to add more special cases.